### PR TITLE
[inductor] fixes for #92397

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #92768
* #92767
* #91952
* #92667
* #92766

### CODE REVIEW

**Comment by nkaretnikov** [comment]
See https://github.com/pytorch/pytorch/pull/92397/files#r1083500462 for context

**Review by ezyang — APPROVED** [review]

**Comment by nkaretnikov** [comment]
I couldn't just import `V` as in the other files because it creates a circular import.

**Comment by nkaretnikov** [comment]
Also: this doesn't fix all the errors for dyn-shapes tests. I still see a bunch of "RuntimeError: Trying to extract a concrete int out of a symbolic int" even with this fix. This just fixes the obvious import/type errors.

File: torch/_inductor/ir.py [hunk 1]
```diff
@@ -128,6 +127,18 @@ def reads_from_conv(buf, var_ranges):
...
 
+def convert_shape_to_symint(lst: List[Union[int, sympy.Expr]]) -> List[torch.SymInt]:
+    """
+    Takes a list of shapes from Inductor and converts them into symints (or just
+    ints if all shapes are static).
+    """
+    if all(isinstance(i, int) for i in lst):
+        return lst
+    if all(isinstance(i, sympy.Integer) for i in lst):
+        return [int(i) for i in lst]
```
Comment (nkaretnikov) [inline torch/_inductor/ir.py:138]: Hmm, I guess my fix for the type signature wasn't correct. Instead, I should have added int to the Union.

**Comment by nkaretnikov** [comment]
No longer needed since https://github.com/pytorch/pytorch/pull/92774 has landed. The SymInt failures mentioned above are fixed by https://github.com/pytorch/pytorch/pull/92609, which wasn't in viable/strict when I rebased. Closing.

**Pull request closed without merging**